server/lib/realtime/rls/subscriptions/subscriptions.ex

```diff
@@ -0,0 +1,260 @@
+                        entities: Map.get(oids, {schema, table}, []),
+                        filters: [],
+                        topic: topic,
+                        user_id: user_id
+                      }
+                      | acc
+                    ]
+
+                  [schema] ->
+                    [
+                      %{
+                        entities: Map.get(oids, {schema}, []),
+                        filters: [],
+                        topic: topic,
+                        user_id: user_id
+                      }
+                      | acc
+                    ]
+                end
+              else
+                acc
+              end
+            end)
+
+          _ ->
+            []
+        end
+
+      {:ok, enriched_params}
+    end)
+  end
+
+  defp generate_topic_subscriptions(%Multi{} = multi) do
+    Multi.run(multi, :valid_topic_subscriptions, fn _,
+                                                    %{
+                                                      enriched_subscription_params:
+                                                        enriched_subscription_params
+                                                    } ->
+      topic_subs =
+        case enriched_subscription_params do
+          [_ | _] ->
+            Enum.reduce(enriched_subscription_params, [], fn %{
+                                                               entities: entities,
+                                                               filters: filters,
+                                                               user_id: user_id
+                                                             },
+                                                             acc ->
+              case entities do
+                [_ | _] ->
+                  [
+                    Enum.reduce(entities, [], fn oid, i_acc ->
+                      %Subscription{}
+                      |> Subscription.changeset(%{
+                        user_id: user_id,
+                        entity: oid,
+                        filters: filters
+                      })
+                      |> case do
+                        %Changeset{changes: topic_sub, valid?: true} -> [topic_sub | i_acc]
+                        _ -> i_acc
+                      end
+                    end)
+                    | acc
+                  ]
+
+                _ ->
+                  acc
+              end
+            end)
+            |> List.flatten()
+
+          _ ->
+            []
+        end
+
+      {:ok, topic_subs}
+    end)
+  end
+
+  defp insert_topic_subscriptions(%Multi{} = multi) do
+    Multi.run(multi, :insert_topic_subscriptions, fn _,
+                                                     %{
+                                                       valid_topic_subscriptions: valid_subs
+                                                     } ->
+      total_inserts =
+        valid_subs
+        |> Enum.chunk_every(20_000)
```

Comment:
there's a postgres limit of 65_535 parameters so chunking until i can come up with a more performant query using `unnest`